pkg/controller/stacks/install/stackinstall_test.go

```diff
@@ -91,6 +91,19 @@ func withStackRecord(stackRecord *corev1.ObjectReference) resourceModifier {
 	return func(r v1alpha1.StackInstaller) { r.SetStackRecord(stackRecord) }
 }
 
+// withPackage allows a test to set a StackInstaller's package
+// Another option would have been to modify the interface to allow this,
+// but it is preferable if we treat the package field as immutable.
+func withPackage(pkg string) resourceModifier {
+	return func(r v1alpha1.StackInstaller) {
+		if si, ok := r.(*v1alpha1.StackInstall); ok {
```

Comment:
`StackInstaller` has `GetPackage`, but not `SetPackage`.  Most of the methods it has today have Get+Set, but we haven't needed SetPackage until now.

You could add `SetPackage` to the interface and implement it for `StackInstall` and `ClusterStackInstall` using either side of your conditional here, and then `withPackage` just calls `SetPackage`.

Reply:
I considered adding `SetPackage`, but I decided not to because I liked the idea of treating more fields as immutable. I also did not like the idea of adding methods to the interface just to make the testing more convenient, so I figured I would do this until we came up with a non-testing use-case for changing the interface.